architecture/2025-02-user-defined-resource-type-feature-spec.md

```diff
@@ -0,0 +1,1055 @@
+
+**Priority: **p2 – Medium (workaround exists)
+
+Radius is not opinionated about which infrastructure as code solution to use. Therefore, it is important that Radius is not biased towards one solution versus another. The parameter `--template-kind` and `--template-path` are biased towards Bicep as Bicep files are called templates and Terraform files are configurations. 
+
+Given this, the `--template-kind` and `--template-path` arguments will be renamed `--recipe-kind` and `--recipe-path`. The previous commands will be retained for backwards compatibility.
+
+## Scenario 3 – Connections
+
+### User Story 8 – Connections into Applications.Core/containers
+
+As a developer, I need to read properties from a connected resource using environment variables injected into my container.
+
+**Priority:** p0 – Critical (core functionality)
+
+**Summary**
+
+All resource in Radius will support connections to other resources, no matter the type. When a connection exists from a connecting resource (destination) to another connected resource (source), the source resource's properties will be available within the destination resource. In the case of `Applications.Core/containers` these properties are manifested as environment variables.
+
+**User Experience**
+
+When a developer adds a connection to a `Applications.Core/containers` resource, environment variables for each of the source resource's properties are created within the environment. 
+
+For example, the developer creates a connection to the PostgreSQL resource type from User Story 1. 
+
+```yaml
+connections: {
+  ordersDB:
+    source: ordersDB.id
+}
+```
+
+The following environment variables are automatically created in the container:
+
+| Environment Variable Name                              | Property Name                              |
+| ------------------------------------------------------ | ------------------------------------------ |
+| `CONNECTIONS_ORDERSDB_PROPERTIES_ENVIRONMENT`          | `ordersDB.properties.environment`          |
+| `CONNECTIONS_ORDERSDB_PROPERTIES_APPLICATION`          | `ordersDB.properties.application`          |
+| `CONNECTIONS_ORDERSDB_PROPERTIES_DATABASE`             | `ordersDB.properties.database`             |
+| `CONNECTIONS_ORDERSDB_PROPERTIES_SIZE`                 | `ordersDB.properties.size`                 |
+| `CONNECTIONS_ORDERSDB_PROPERTIES_HOST`                 | `ordersDB.properties.host`                 |
+| `CONNECTIONS_ORDERSDB_PROPERTIES_PORT`                 | `ordersDB.properties.port`                 |
+| `CONNECTIONS_ORDERSDB_PROPERTIES_CREDENTIALS_USERNAME` | `ordersDB.properties.credentials.username` |
+| `CONNECTIONS_ORDERSDB_PROPERTIES_CREDENTIALS_PASSWORD` | `ordersDB.properties.credentials.password` |
+
+If the developer prefers for these environment variables to not be created, he/she can set the existing `disableDefaultEnvVars` boolean property to TRUE.
+
+### User Story 9 – Connections into recipes
```

Comment:
Do we have a third scenario for UDT to UDT connections, or does this cover that?